web-client/src/pages/profile/[id].tsx

```diff
@@ -16,17 +14,16 @@ const ViewPage = () => {
       const response = await http.get(`http://localhost:4000/v1/users/${id}`);
 
       setUser(response.data.data);
-    } catch (err: unknown) {
-      if (err instanceof AxiosError) {
-        const status = err.response?.status;
-        const errorData = err.response?.data;
-
-        // The console.log stays while the error isn't properly annotated
-        console.log(errorData);
+    } catch (e: unknown) {
+      if ((e instanceof AxiosError) && e.response) {
+        const err: HttpResponse<ServerAPI['UserError']> = e.response;
```

Comment:
There's no need to pack multiple things into the same error type (`UserError`). These things cost nothing.

But also, for this specific situation, you can also just return a status code 404 from the server which is a perfect fit (object does not exist).